components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/LocalAuthenticatorConfig.java

```diff
@@ -111,8 +118,17 @@ public static LocalAuthenticatorConfig build(OMElement localAuthenticatorConfigO
                     Property[] propertiesArr = propertiesArrList.toArray(new Property[0]);
                     localAuthenticatorConfig.setProperties(propertiesArr);
                 }
+            } else if ("DefinedBy".equals(member.getLocalName())) {
+                localAuthenticatorConfig.setDefinedByType(DefinedByType.valueOf(member.getText()));
             }
         }
+
+        if (localAuthenticatorConfig.getDefinedByType() == null) {
+            localAuthenticatorConfig.setDefinedByType(DefinedByType.SYSTEM);
+            LOG.debug("The defined by type is not set for the {}. Hence setting default SYSTEM value.",
+                    localAuthenticatorConfig.getName());
+        }
+
```

Comment:
IMO this is redundant

Reply:
Addressed with commit fe95d4b60aeb8557a6ac7cfe51ce2576ee94112c